### PR TITLE
Nebraska Child Care Subsidy

### LINKED ISSUE #4845: Nebraska Child Care Subsidy
https://nebraskalegislature.gov/laws/statutes.php?statute=68-1206

https://dhhs.ne.gov/Pages/Child-Care-Parents.aspx

### PR DESCRIPTION
**Pull request by vrathi101**
Fixes #4845

### CODE REVIEW

File: policyengine_us/variables/gov/states/ne/dhhs/ne_child_care_subsidy.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from policyengine_us.model_api import *
+
+
+class ne_child_care_subsidy(Variable):
+    value_type = float
+    entity = Person
+    label = "Amount of Nebraska Child Care Subsidy program benefit"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/ne_child_care_subsidy.py:7]:
```suggestion
    label = "Nebraska Child Care Subsidy"
```

File: policyengine_us/variables/gov/states/ne/dhhs/ne_child_care_subsidy.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from policyengine_us.model_api import *
+
+
+class ne_child_care_subsidy(Variable):
+    value_type = float
+    entity = Person
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/ne_child_care_subsidy.py:6]: person or tax unit?

File: policyengine_us/parameters/gov/states/ne/dhhs/fpg_fraction.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+values:
+  2021-08-28: 1.30
+  2023-01-01: 1.85
+  2026-10-01: 1.30
+
+metadata:
+  unit: /1
+  period: year
+  label: Nebraska child care assistance federal poverty guideline limit
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/dhhs/fpg_fraction.yaml:11]:
```suggestion
  label: Nebraska child care subsidy federal poverty guideline limit
```

File: policyengine_us/parameters/gov/states/ne/dhhs/fpg_fraction.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+description: Nebraska provides child care assistance to families with incomes at or below this fraction of the federal poverty guideline.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/dhhs/fpg_fraction.yaml:1]:
```suggestion
description: Nebraska provides the child care subsidy to families with income at or below this fraction of the federal poverty guidelines.  
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ne/dhhs/fpg_fraction.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+description: Nebraska provides the child care subsidy to families with income at or below this fraction of the federal poverty guidelines.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/dhhs/fpg_fraction.yaml:1]:
```suggestion
description: Nebraska limits the child care subsidy to families with income at or below this fraction of the federal poverty guidelines.  
```

File: policyengine_us/parameters/gov/states/ne/dhhs/fpg_fraction.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/dhhs/fpg_fraction.yaml:1]: lets make another folder in dhhs/child_care_subsidy/

File: policyengine_us/variables/gov/states/ne/dhhs/ne_child_care_subsidy_eligible.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+from policyengine_us.model_api import *
+
+
+class ne_child_care_subsidy_eligible(Variable):
+    value_type = bool
+    entity = TaxUnit
+    label = "Nebraska Child Care Subsidy program eligible"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/ne_child_care_subsidy_eligible.py:7]:
```suggestion
    label = "Eligible for the Nebraska Child Care Subsidy program"
```

File: policyengine_us/variables/gov/states/ne/dhhs/ne_child_care_subsidy.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+from policyengine_us.model_api import *
+
+
+class ne_child_care_subsidy(Variable):
+    value_type = float
+    entity = TaxUnit
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/ne_child_care_subsidy.py:6]: This might need to be SPMUnit

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/fpg_fraction.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+description: Nebraska limits the child care subsidy to families with income at or below this fraction of the federal poverty guidelines.  
+
+values:
+  2021-08-28: 1.30
+  2023-01-01: 1.85
+  2026-10-01: 1.30
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/fpg_fraction.yaml:6]:
```suggestion
  2021-08-28: 1.3
  2023-01-01: 1.85
  2026-10-01: 1.3
```

File: policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/special_need_age_threshold.yaml [hunk 9]
```diff
@@ -0,0 +1,13 @@
+description: Nebraska limits the child care subsidy to children this age or younger, given they do require care due to a special need.  
+values:
+  2021-08-28: 19
+metadata:
+  period: year
+  unit: year
+  label: Nebraska child care subsidy special need child age threshold
+  reference:
+    # Age threshold not mentioned in legal code
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/special_need_age_threshold.yaml:9]: Does the legal code point to a different program?

File: policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy_eligible.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    label = "Eligible for the Nebraska Child Care Subsidy program"
+    definition_period = YEAR
+    reference = (
+        "https://nebraskalegislature.gov/laws/statutes.php?statute=68-1206",
+        "https://dhhs.ne.gov/Pages/Child-Care-Subsidy-Information-for-Parents.aspx",
+    )
+    defined_for = StateCode.NE
+
+    def formula(spm_unit, period, parameters):
+        print("\ntestcase")
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy_eligible.py:16]: remove prints once done

File: policyengine_us/variables/household/demographic/person/is_student.py [hunk 11]
```diff
@@ -0,0 +1,13 @@
+from policyengine_us.model_api import *
+
+
+class is_student(Variable):
+    value_type = bool
+    entity = Person
+    label = "Is a part time student"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/demographic/person/is_student.py:7]:
```suggestion
    label = "Is a part or full time student"
```

File: policyengine_us/variables/household/demographic/person/is_student.py [hunk 12]
```diff
@@ -0,0 +1,13 @@
+from policyengine_us.model_api import *
+
+
+class is_student(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/demographic/person/is_student.py:4]: does this include college and high-school

File: policyengine_us/variables/household/demographic/person/has_special_needs.py [hunk 13]
```diff
@@ -0,0 +1,8 @@
+from policyengine_us.model_api import *
+
+
+class has_special_needs(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/demographic/person/has_special_needs.py:4]: is there a legal code definition? And this specific for this program (are there special rules which make you "special needs" for this program)

File: policyengine_us/variables/household/demographic/person/goes_to_medical_or_therapy_visits.py [hunk 14]
```diff
@@ -0,0 +1,8 @@
+from policyengine_us.model_api import *
+
+
+class goes_to_medical_or_therapy_visits(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/demographic/person/goes_to_medical_or_therapy_visits.py:4]: same question as for special needs

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/household/demographic/person/is_student.py (hunk 12 above)
Comment (vrathi101) [reply policyengine_us/variables/household/demographic/person/is_student.py:4]: yes, included in `is_full_time_student` variable

File: policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/special_need_age_threshold.yaml (hunk 9 above)
Comment (vrathi101) [reply policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/special_need_age_threshold.yaml:9]: no oddly legal code only mentions income eligibility, none of the other parameters added (which I found in Nebraska DHHS sources)

File: policyengine_us/variables/household/demographic/person/has_special_needs.py (hunk 13 above)
Comment (vrathi101) [reply policyengine_us/variables/household/demographic/person/has_special_needs.py:4]: i found another source that has more details (which is NE specific) so I will move variable location

File: policyengine_us/variables/household/demographic/person/goes_to_medical_or_therapy_visits.py (hunk 14 above)
Comment (vrathi101) [reply policyengine_us/variables/household/demographic/person/goes_to_medical_or_therapy_visits.py:4]: not sure for this case (and for other variables I added in categorical.yaml that relate to parent eligibility): I wasn't able to find specific definitions, only could find https://dhhs.ne.gov/Pages/Child-Care-Subsidy-Information-for-Parents.aspx
image

Comment (PavelMakarchuk) [reply policyengine_us/variables/household/demographic/person/goes_to_medical_or_therapy_visits.py:4]: Lets make this one general variable that combines all of these definitions - dont need one for each - empty input variable

File: policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/special_need_age_threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/special_need_age_threshold.yaml:1]: lets a age_threshold folder and create two files; `special_needs` and `base`

File: policyengine_us/variables/household/expense/childcare/childcare_expenses.py [hunk 15]
```diff
@@ -13,9 +13,13 @@ def formula(spm_unit, period, parameters):
...
         )
         # States where we model childcare subsidies.
-        STATES_WITH_CHILD_CARE_SUBSIDIES = ["CA", "CO"]
+        STATES_WITH_CHILD_CARE_SUBSIDIES = ["CA", "CO", "NE"]
         subsidy_variables = [
-            i.lower() + "_child_care_subsidies"
+            (
+                i.lower() + "_child_care_subsidy"
+                if i == "NE"
+                else i.lower() + "_child_care_subsidies"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/expense/childcare/childcare_expenses.py:21]: might want to consolidate this by creating a general `ne_child_care_subsidies.py` - not sure if there are other subsidy programs

File: policyengine_us/parameters/gov/household/household_state_benefits.yaml [hunk 16]
```diff
@@ -6,6 +6,7 @@ values:
     # State child care subsidies.
     - ca_child_care_subsidies
     - co_child_care_subsidies
+    - ne_child_care_subsidies
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/household/household_state_benefits.yaml:9]: while you are here, could you separate the programs by state with a comment?

File: policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/age_threshold/special_needs.yaml [hunk 17]
```diff
@@ -0,0 +1,13 @@
+description: Nebraska limits the child care subsidy to children this age or younger, given they require care due to a special need.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/age_threshold/special_needs.yaml:1]:
```suggestion
description: Nebraska limits the child care subsidy to children of this age or younger, requiring special needs care.  
```

File: policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/rate.yaml [hunk 18]
```diff
@@ -0,0 +1,15 @@
+description: Nebraska obligates families participating in child care subsidy to pay this fraction of their income, given their income exceeds the federal poverty guideline fraction threshold.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/rate.yaml:1]:
```suggestion
description: Nebraska obligates families participating in the child care subsidy program to pay childcare expenses of this fraction of income excess over federal poverty guidelines. 
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy_eligible.py [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+    )
+    defined_for = StateCode.NE
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.ne.dhhs.child_care_subsidy
+        person = spm_unit.members
+        eligible_parents = spm_unit.sum(
+            spm_unit.members("ne_child_care_subsidy_eligible_parent", period)
+        )
+        has_eligible_parents = eligible_parents > 0
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy_eligible.py:21]:
```suggestion
        person = spm_unit.members
        eligible_parent = person("ne_child_care_subsidy_eligible_parent", period)
        eligible_parent_present = spm_unit.any(eligible_parent)
```

File: policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy_eligible.py [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+        eligible_children = spm_unit.sum(
+            spm_unit.members("ne_child_care_subsidy_eligible_child", period)
+        )
+        has_eligible_children = eligible_children > 0
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy_eligible.py:25]:
```suggestion
        eligible_child = person("ne_child_care_subsidy_eligible_child", period)
        eligible_child_present = spm_unit.any(eligible_child)
```

File: policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy_eligible.py [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+    definition_period = YEAR
+    reference = (
+        "https://nebraskalegislature.gov/laws/statutes.php?statute=68-1206",
+        "https://dhhs.ne.gov/Pages/Child-Care-Subsidy-Information-for-Parents.aspx",
+    )
+    defined_for = StateCode.NE
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.ne.dhhs.child_care_subsidy
+        person = spm_unit.members
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy_eligible.py:17]:
```suggestion
```

File: policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy.py [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+    defined_for = "ne_child_care_subsidy_eligible"
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.ne.dhhs.child_care_subsidy
+        childcare_expenses = spm_unit(
+            "spm_unit_pre_subsidy_childcare_expenses", period
+        )
+        income = add(spm_unit, period, ["adjusted_gross_income"])
+        fpg = add(spm_unit, period, ["tax_unit_fpg"])
+        fee_obligated = income > fpg * p.fpg_fraction.threshold
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy.py:18]:
```suggestion
        income_above_fpg_fraction = income > fpg * p.fpg_fraction.threshold
```

File: policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy.py [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+        capped_childcare_expenses_with_fee = max_(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy.py:19]: separate out the mutiplication

File: policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy.py [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+    definition_period = YEAR
+    defined_for = "ne_child_care_subsidy_eligible"
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.ne.dhhs.child_care_subsidy
+        childcare_expenses = spm_unit(
+            "spm_unit_pre_subsidy_childcare_expenses", period
+        )
+        income = add(spm_unit, period, ["adjusted_gross_income"])
+        fpg = add(spm_unit, period, ["tax_unit_fpg"])
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy.py:17]: use spm_unit_fpg

File: policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy.py [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+    label = "Nebraska Child Care Subsidy"
+    definition_period = YEAR
+    defined_for = "ne_child_care_subsidy_eligible"
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.ne.dhhs.child_care_subsidy
+        childcare_expenses = spm_unit(
+            "spm_unit_pre_subsidy_childcare_expenses", period
+        )
+        income = add(spm_unit, period, ["adjusted_gross_income"])
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy.py:16]: if we need the income of the spm_unit then we can use `spm_unit_net_income`

File: policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy.py [hunk 26]
```diff
@@ -0,0 +1,26 @@
+from policyengine_us.model_api import *
+
+
+class ne_child_care_subsidy(Variable):
+    value_type = float
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/dhhs/child_care_subsidy/ne_child_care_subsidy.py:5]:
```suggestion
    value_type = float
    unit = USD
```

File: policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/rate.yaml [hunk 27]
```diff
@@ -0,0 +1,15 @@
...
+
+values:
+  2021-08-28: 0.07
+
+metadata:
+  unit: /1
+  period: year
+  label: Nebraska child care subsidy family fee rate
+  reference:
+    # Child care subsidy family fee not mentioned in legal code
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/dhhs/child_care_subsidy/rate.yaml:11]: remove then, most benefit programs just online manuals

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**